.github/workflows/docker-slurm.yml

```diff
@@ -0,0 +1,130 @@
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      -
+        name: Build frontend
+        uses: docker/build-push-action@v4
+        with:
+          context: ./docker
+          file: ./docker/frontend/Dockerfile
+          push: false
+          load: true
+          tags: ghcr.io/noaa-gsl/exascaleworkflowsandbox/frontend:latest
+          cache-from: type=registry,ref=ghcr.io/noaa-gsl/exascaleworkflowsandbox/frontend:cache,mode=max
+          cache-to: type=registry,ref=ghcr.io/noaa-gsl/exascaleworkflowsandbox/frontend:cache,mode=max
+      -
+        name: Build master
+        uses: docker/build-push-action@v4
+        with:
+          context: ./docker
+          file: ./docker/master/Dockerfile
+          push: false
+          load: true
+          tags: ghcr.io/noaa-gsl/exascaleworkflowsandbox/master:latest
+          cache-from: type=registry,ref=ghcr.io/noaa-gsl/exascaleworkflowsandbox/master:cache,mode=max
+          cache-to: type=registry,ref=ghcr.io/noaa-gsl/exascaleworkflowsandbox/master:cache,mode=max
+      -
+        name: Build node
+        uses: docker/build-push-action@v4
+        with:
+          context: ./docker
+          file: ./docker/node/Dockerfile
+          push: false
+          load: true
+          tags: ghcr.io/noaa-gsl/exascaleworkflowsandbox/node:latest
+          cache-from: type=registry,ref=ghcr.io/noaa-gsl/exascaleworkflowsandbox/node:cache,mode=max
+          cache-to: type=registry,ref=ghcr.io/noaa-gsl/exascaleworkflowsandbox/node:cache,mode=max
+      -
+        name: Start Slurm cluster containers
+        run: |
+          cd docker
+          export DOCKER_CLIENT_TIMEOUT=600
+          export COMPOSE_HTTP_TIMEOUT=600
+          docker-compose -f docker-compose.yml up -d --no-build
+      -
+        name: Create ssh keys in cluster
+        run: |
+          docker exec frontend ssh-keygen -t rsa -f /home/admin/.ssh/id_rsa -N ""
+          docker exec frontend cp /home/admin/.ssh/id_rsa.pub /home/admin/.ssh/authorized_keys
+      -
+        name: Copy test scripts to work directory
+        run: |
+          docker exec frontend bash -l -c "mkdir -p work; cd work ; cp ../test/*.py . ; cp ../test/*.f90 ."
+      -
+        name: Parsl hello test
+        run: |
+          docker exec frontend bash -l -c "spack env activate flux; cd work ; ./parsl_hello.py"
+      -
+        name: Parsl/Flux hello MPI test
+        run: |
+          docker exec frontend bash -l -c "spack env activate flux; cd work ; ./parsl_mpi_flux_hello.py"
+          docker exec frontend bash -l -c "cat work/resource_list.txt"
+          docker exec frontend bash -l -c "cat work/mpi_apps.hello.out"
+# Uncomment this step to enable interactive debugging
+#      -
+#        name: Debug session
+#        if: ${{ failure() }}
+#        uses: mxschmitt/action-tmate@v3
+#        timeout-minutes: 60
+#        with:
+#          limit-access-to-actor: true
```

Comment:
Confirming that leaving this in is the right option. Could we potentially move this code block to its only file? How will the user interactive with this code block? Will this require the user to uncomment this section or is this just for internal use?

Reply:
This step is only useful when the github actions is failing and the reason cannot be determined without gaining interactive access to the runner for debugging purposes.  The reason I leave it in here commented out is because it's a recipe that isn't at all obvious and which will likely be needed in the future due to complexities of Slurm/Parsl/Flux.  I could remove it, but then, I'd have to remember it from scratch the next time it is needed or store it somewhere else.  Although it's not usually a good idea to leave commented code, this is a GHA workflow step that is likely to be needed repeatedly in the future and it's much simpler to just leave it here.